src/nvme/ioctl.h

```diff
@@ -2157,12 +2159,66 @@ static inline int nvme_get_log_lockdown(nvme_link_t l,
 /**
  * nvme_set_features() - Set a feature attribute
  * @l:		Link handle
- * @args:	&struct nvme_set_features_args argument structure
+ * @data:	User address of feature data, if applicable
+ * @nsid:	Namespace ID, if applicable
+ * @cdw11:	Value to set the feature to
+ * @cdw12:	Feature specific command dword12 field
+ * @cdw13:	Feature specific command dword13 field
+ * @cdw15:	Feature specific command dword15 field
+ * @data_len:	Length of feature data, if applicable, in bytes
+ * @save:	Save value across power states
+ * @uuidx:	UUID Index for differentiating vendor specific encoding
+ * @fid:	Feature identifier
+ * @result:	The command completion result from CQE dword0
  *
  * Return: 0 on success, the nvme command status if a response was
  * received (see &enum nvme_status_field) or a negative error otherwise.
  */
-int nvme_set_features(nvme_link_t l, struct nvme_set_features_args *args);
+static inline int nvme_set_features(nvme_link_t l, void *data, __u32 nsid,
+				     __u32 cdw11, __u32 cdw12, __u32 cdw13,
+				     __u32 cdw15, __u32 data_len, bool save,
+				     __u8 uuidx, __u8 fid, __u32 *result)
```

Comment:
Could you reorder these arguments (see comment in https://github.com/linux-nvme/libnvme/issues/495). And also rename uuidx do `uidx` as it is in the spec? My aim is to follow the spec as closely as possible.

Reply:
Done :)
I went through all functions I changed and checked against the specs. Please let me know if you spot another mismatch :)